gst/datarepo/gstdatareposrc.c

```diff
@@ -291,6 +270,10 @@ gst_data_repo_src_finalize (GObject * object)
   if (src->shuffled_index_array)
     g_array_free (src->shuffled_index_array, TRUE);
 
+  /* Check for gst-inspect log */
+  if (src->caps)
+    gst_caps_unref (src->caps);
```

Comment:
How do you sure that the ref-count will become 0 with a single call of unref?
```gst_data_repo_src_get_caps``` may have been called multiple times.

Reply:
Thank, you.
The ref count increased in gst_data_repo_src_get_caps is decreased in gstbasesrc.c after the function returns.

Reply:
Ok. if we can sure that it will become ref=0 after this unref, please set it NULL after unref. (to protect it from further processing with other places where we might need to unref it)

Reply:
ok, thank you.